example_school_library_decorator/functions.rb

```diff
@@ -0,0 +1,17 @@
+def
+  list_books(books, _user_select)
+  book_list = ['Rapunzel', 'Oliver Twist', 'Othello', 'Romeo and Juliette']
+  people = %w[John Isaac Emma Victor]
+  while userSelect
+    case books
+    when 1
+      puts 'The following is a list of books'
+      puts book_list
+      break
+    when 2
+      puts 'The following is a list of people'
+      puts people
+      break
+    end
+  end
+end
```

Comment:
- [ ] Let's look back at the project requirements in the quote below:

> - Create a `main.rb` file that will serve as your console app entry-point. It should have methods that do the following:
>   - List all books.
>   - List all people.
>   - Create a person (teacher or student, not a plain `Person`).
>   - Create a book.
>   - Create a rental.
>   - List all rentals for a given person id.

Since this is where you are defining the methods for `main.rb`, please ensure that the methods listed above are implemented. Right now all the above desire operations cannot be performed in the app.

## Suggestions

It would have been nice to have all methods defined in a Class, that would have made your codebase to have organized logic. All the same, consider these points I am about to share with you.

1. The app will be creating books, it would be nice to have an array to store the added books. Likewise with people and rentals 👇🏼:

```ruby
books = []    
people = []    
rentals = []
```
2. Next, create methods that will iterate the above arrays and display their data in the console:
```ruby
def list_books
  # logic for listing all books
end

def list_people
  # logic for listing people
end

def list_rentals(person_id)
  # logic for listing rentals by one person filtering by person id
end
  
```

3. Next, create methods that will help a user create a person, book, and rental.

```ruby
def create_person
  # logic for creating a person
  # ensure the created person is pushed to the persons array
end

def create_book
  # logic for creating a book
  # ensure the created book is pushed to the books array
end
  
def create_rental
  # logic for creating rental
  # ensure the created rental is pushed to the rentals array
end
```
These methods for creating data should be interactive. They should be able to ask users to key in data for given object data values. The keyed-in data should then be used to create the intended object i.e. a book.